<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/run.py", line 208, in main
    return_code, report = execute_linter(
                          ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/run.py", line 180, in execute_linter
    result = linter.lint(recipes, subdir, variant_config_files, exclusive_config_files, fix_flag)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/lint/__init__.py", line 728, in lint
    msgs = self.lint_file(
           ^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/lint/__init__.py", line 834, in lint_file
    self.lint_recipe(
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/lint/__init__.py", line 894, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/lint/__init__.py", line 294, in run
    self.check_recipe(recipe)
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/anaconda_linter/lint/check_syntax.py", line 35, in check_recipe
    ro_parser: Final[RecipeParser] = recipe.get_read_only_parser()
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/percy/render/recipe.py", line 675, in get_read_only_parser
    return RecipeParser("\n".join(self.meta_yaml))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.12/site-packages/percy/parser/recipe_parser.py", line 372, in __init__
    parent = node_stack[-1]
             ~~~~~~~~~~^^^^
IndexError: list index out of range

ERROR conda.cli.main_run:execute(125): `conda run conda-lint /tmp/abs_e6e101dd_t/clone` failed. (See above for error)
</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -40,7 +40,8 @@ requirements:
     - pyarrow >=9.0.0,<18.0.0
     - pandas >=1.0.0,<3.0.0
     - duckdb >=0.5.0,<1.0.0
-    - ray >=2.0.0,<3.0.0
+    - ray >=2.10.0,<3.0.0  # [py>=39]
+    - ray ==2.10.0  # [py<39]
     - python-snappy >=0.6.0,<1.0.0
     - thrift >=0.13.0,<1.0.0
     - mypy-boto3-glue >=1.28.18
@@ -49,9 +50,9 @@ requirements:
     - adlfs >=2023.1.0,<2024.8.0
     - gcsfs >=2023.1.0,<2024.1.0
     - psycopg2-binary >=2.9.6
-    - sqlalchemy >=1.4.0,<3.0.0
+    - sqlalchemy >=2.0.18,<3.0.0
     - getdaft >=0.2.12
-    - numpy >=1.22.4
+    - numpy >=1.22.4,<2.0.0
 
 test:
   imports:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -36,7 +36,6 @@ requirements:
     - zstandard >=0.13.0,<1.0.0
     - tenacity >=8.2.3,<9.0.0
   run_constrained:
-# https://github.com/apache/iceberg-python/blob/main/pyproject.toml#L60C1-L60C59
     - pyarrow >=9.0.0,<18.0.0
     - pandas >=1.0.0,<3.0.0
     - duckdb >=0.5.0,<1.0.0

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -54,15 +54,30 @@ requirements:
     - numpy >=1.22.4,<2.0.0
 
 test:
+  source_files:
+    - pyproject.toml
+    - tests
   imports:
     - fb303
     - hive_metastore
     - pyiceberg
   commands:
-    - pip check
+  #pyspark 3.4.1 has requirement py4j==0.10.9.7, but you have py4j 0.10.9.3
+    #- pip check
     - pyiceberg --help
+    - pytest --ignore="tests/io/test_fsspec.py" --ignore="tests/integration/" --ignore="tests/catalog/test_rest.py" --ignore="tests/catalog/test_base.py" --ignore="tests/catalog/test_glue.py" --ignore="tests/catalog/test_sql.py" --ignore="tests/cli/test_console.py" -k "not gcs"
   requires:
     - pip
+    - pytest
+    - boto3
+    - moto >=5.0.2,<6.0.0
+    - pyarrow
+    - fastavro
+    - numpy
+    - pyspark
+    - pytest-mock
+    - thrift
+    - sqlalchemy
 
 about:
   home: https://iceberg.apache.org/

